packages/plugin-infra/src/types.ts

```diff
@@ -133,8 +134,8 @@ export namespace Plugin.Shared {
         web3?: Web3EnableRequirement
     }
     export interface Web3EnableRequirement {
-        /** This flag indicates the plugin entry in the composition entry should be hidden if the current chain is invalid. */
-        compositionEntryRequiresChainIDValid?: boolean
+        /** Plugin can declare what chain it supports. When the current chain is not supported, the composition entry will be hidden. */
+        compositionEntryRequiredChains?: ChainId[]
```

Comment:
Highly request `postInspectorRequiredChains` too! And add a switch network panel when the chosen chain is not matched with the expected.

Reply:
In the original design, it did have this kind of API, but the problem is, some plugins are rendering their own "Chain not supported" UI, therefore I don't know how to handle this case.

Reply:
And another problem is, the current abstraction level doesn't support detecting if the plugin needs to render without actually render the content. I'll add two new APIs later that might resolve this problem.